src/hash-helper.js

```diff
@@ -0,0 +1,16 @@
+import crypto from 'crypto';
+
+export function getHasher(compiler = null) {
+  const hashFunction =
+    compiler && compiler.output && compiler.output.hashFunction;
+
+  if (typeof hashFunction === 'string') {
+    return crypto.createHash(hashFunction);
+  } else if (typeof hashFunction === 'function') {
+    return hashFunction();
+  } 
+    return crypto.createHash('md4');
+  
+}
+
+export default getHasher;
```

Comment:
Please put it as the static method for TerserPlugin class

Reply:
Done. Note that this seemed a bit strange to me since it does create a circular dependency, but shouldn't be a problem in reality.